### ISSUE TITLE
Malformed JSON-LD

### ISSUE THREAD

**Issue by gregordonsky** [Yoast/Yoast-SEO-for-TYPO3#412]
### Please give us a description of what happened.

The JSON-LD is malformed. There is a ; character at the end of <script type="application/ld+json"> and before </script>
If you use facebook pixel in the browser console you see this warning:
[Facebook Pixel] - Unable to parse JSON-LD tag. Malformed JSON

### Please describe what you expected to happen and why.

No warning in the browser console. 

### Additional context
This pull request changed the previous behaviour:
#354 
I didn't find any example with a semicolon at the end of structured data; however i don't know if this is a correct practice.

Please remove the ; character at:
https://github.com/Yoast/Yoast-SEO-for-TYPO3/blob/968ba5be9b12d818ea82b42b5d4574787a03c0c6/Classes/StructuredData/StructuredDataProviderManager.php#L87
or give the possibility to manage the presence of the semicolon.

### Technical info
* TYPO3 version: 9.5.24
* Yoast SEO version: 7.1.3

**Comment by swiftyone** [comment]
Hi @gregordonsky, you can fix this issue by overriding the StructuredDataProviderManager:

Add this to your Extensions ext_localconf.php

```php
$GLOBALS['TYPO3_CONF_VARS']['SC_OPTIONS']['t3lib/class.t3lib_pagerenderer.php']['render-postProcess'][]
            = \Vendor\Extension\DataProvider\Seo\StructuredDataProviderManager::class . '->render';
```

And add this class:

```php
<?php

declare(strict_types=1);

namespace Vendor\Extension\DataProvider\Seo;

use YoastSeoForTypo3\YoastSeo\StructuredData\StructuredDataProviderManager as BaseStructuredDataProviderManager;

/**
 * Class StructuredDataProviderManager.
 */
class StructuredDataProviderManager extends BaseStructuredDataProviderManager
{
    /**
     * @param array $src
     *
     * @return string
     */
    protected function buildJsonLdBlob($src): string
    {
        $data = [];
        foreach ($src as $provider => $dataItems) {
            foreach ($dataItems as $item) {
                $data[] = $item;
            }
        }

        if (empty($data)) {
            return '';
        }

        return '<script type="application/ld+json">' . json_encode($data, JSON_UNESCAPED_SLASHES) . '</script>';
    }
}

```

**Comment by sal-lochbaum** [comment]
Thanks for the workaround. But that semicolon is just pain wrong there and it should really be removed again.

**Comment by chludwig** [comment]
I was just running into the same issue.  

Please remove the `;` at the end of the json. See [W3C Recommendation for embeding JSON-LD code into html documents](https://www.w3.org/TR/json-ld/#embedding-json-ld-in-html-documents)

The mentioned fix https://github.com/Yoast/Yoast-SEO-for-TYPO3/pull/354 for MATE extension should be done in MATE itself while combining the JSON-LD script tags.

**Comment by RinyVT** [comment]
Closing this issue, the fix has been implemented in the new version 8.0
